starport_template/lib/pages/create_wallet_page.dart

```diff
@@ -8,12 +15,100 @@ class CreateWalletPage extends StatefulWidget {
 }
 
 class _CreateWalletPageState extends State<CreateWalletPage> {
+  Either<BiometricCredentialsStorageFailure, Unit>? _authenticationResult;
+
+  bool get isLoading => _authenticationResult == null;
+
+  bool get isAuthenticating => _authenticationResult == null;
+
+  @override
+  void initState() {
+    super.initState();
+    _authenticateUser();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return const Scaffold(
-      body: Center(
-        child: Text("Create Wallet"),
+    return Scaffold(
+      body: ContentStateSwitcher(
+        isLoading: isLoading,
+        errorChild: _errorUI(),
+        isError: _authenticationResult?.isRight() == false,
+        loadingChild: LoadingSplash(
+          text: isAuthenticating ? 'Authenticating..' : 'Creating a recovery phrase..',
+        ),
+        contentChild: Scaffold(
+          body: _contentUI(),
+          appBar: _appBar(),
+        ),
+      ),
+    );
+  }
+
+  CosmosAppBar _appBar() {
+    return CosmosAppBar(
+      leading: const CosmosBackButton(text: "Back"),
+      title: "Back up your account",
+      actions: [
+        CosmosAppBarAction(
+          onTap: _onTapAdvanced,
+          text: "Advanced",
+        ),
+      ],
+    );
+  }
+
+  Widget _contentUI() {
+    return SafeArea(
+      child: Padding(
+        padding: EdgeInsets.only(
+          left: CosmosTheme.of(context).spacingL,
+          right: CosmosTheme.of(context).spacingL,
+          top: CosmosTheme.of(context).spacingS,
+          bottom: CosmosTheme.of(context).spacingXL,
+        ),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: [
+            Text(
+              "If your device is lost or stolen, you will be able to recover your account.",
+              style: CosmosTextTheme.copy0Normal,
+            ),
+            SizedBox(height: CosmosTheme.of(context).spacingXL),
+            const InfoCard(text: "We will never ask you to share your recovery phrase."),
+            SizedBox(height: CosmosTheme.of(context).spacingL),
+            const InfoCard(text: "Never share your recovery phrase with anyone, store it securely."),
+            SizedBox(height: CosmosTheme.of(context).spacingL),
+            const InfoCard(
+              text: "If you don’t backup your wallet or lose your recovery phrase, "
+                  "you will not able to recover your account",
+            ),
+            const Spacer(),
+            CosmosElevatedButton(text: "Back up now", onTap: _onTapBackUpNow),
+            const SizedBox(height: 10),
```

Comment:
Can we use spacing from `CosmosTheme.of(context)` here?

Reply:
yup! absolutely